Ripples MPI Compilation Error

It seems Ripples build on Perlmutter with MPI enabled would fail on this line. Ripples with MPI disabled will build successfully. The latest good commit hash if MPI is enabled for Ripples: `c6446e291dceb`. g++ version: `g++ (GCC) 11.2.0 20210728 (Cray Inc.)`. Error message:


```
../../tools/mpi-imm.cc: In function ‘int main(int, char**)‘:
../../tools/mpi-imm.cc:166:24: error: expected unqualified-id before ‘=’ token
 166 |  std::allocator<char> = GraphAllocator;
   |            ^
../../tools/mpi-imm.cc:167:62: error: ‘GraphAllocator’ was not declared in this scope
 167 |  GraphFwd Gf = ripples::loadGraph<GraphFwd>(CFG, weightGen, GraphAllocator);
   |                               ^~~~~~~~~~~~~~
Waf: Leaving directory `/global/u1/h/hxu398/ripples/build/release’
Build failed
 -> task in ‘mpi-imm’ failed with exit status 1 (run with -v to display more information)
```

_Originally posted by @hxu296 in https://github.com/pnnl/ripples/pull/6#discussion_r1316290722_

--- a/tools/mpi-imm.cc
+++ b/tools/mpi-imm.cc
@@ -163,8 +163,7 @@ int main(int argc, char *argv[]) {
   }
   GraphBwd &G(*Gr);
 #else
-  std::allocator<char> = GraphAllocator;
-  GraphFwd Gf = ripples::loadGraph<GraphFwd>(CFG, weightGen, GraphAllocator);
+  GraphFwd Gf = ripples::loadGraph<GraphFwd>(CFG, weightGen);
   GraphBwd G = Gf.get_transpose();
 #endif
   console->info("Loading Done!");

